File: app/src/main/java/tech/thdev/app/view/presenter/MainContract.java
package tech.thdev.app.view.presenter;

import android.content.Context;

import tech.thdev.base.presenter.BasePresenter;
import tech.thdev.base.presenter.BaseView;

/**
 * Created by tae-hwan on 18/12/2016.
 */

public interface MainContract {

    interface View extends BaseView {

        void showStartOverlay();

        void showObtainingPermissionOverlayWindow();
    }

    interface Presenter extends BasePresenter<View> {

        void startOverlayWindowService(Context context);

        void onOverlayResult(Context context);
    }
}

File: app/src/main/java/tech/thdev/app/view/presenter/MainContract.kt
package tech.thdev.app.view.presenter

import android.content.Context

/**
 * Created by tae-hwan on 18/12/2016.
 */
interface MainContract {
    interface View {
        fun showStartOverlay()
        fun showObtainingPermissionOverlayWindow()
    }

    interface Presenter {
        var view: MainContract.View

        fun startOverlayWindowService(context: Context?)
        fun onOverlayResult(context: Context?)
    }
}